### DISCUSSION TITLE
WalletConnect V2 Getting No matching key. history: 1692426096597151 error in React+wagmi

### DISCUSSION THREAD

**Discussion by Pavan-0xblock** [wevm/wagmi#2920] [Q&A]
I there,

I'm trying to add WalletConnect V2 in my current running React JS project and I added React JS popup wallet connect modal and it's working fine fore connection my wallet using WalletConnect. And for the contract read and write I'm using @wagmi/core because I can made a global dynamic function it's working using @wagmi/core. It's work's for read and write function but when I wall write function which is using @wagmi/core at that time my popup modal opened as expected in meta mask but after click on Confirm Or Reject it gives me errors like No matching key. history: 1692426096597151 as attached in snapshot.

![image (1)](https://github.com/wagmi-dev/wagmi/assets/100664726/ea093ac7-80b6-498e-8ec3-9f15c22e7e98)
![IMG_4756](https://github.com/wagmi-dev/wagmi/assets/100664726/120b7a4f-8161-4dc1-9167-858782350c26)

**Comment by glitch-txs** [comment]
Hi! do you have a MRE to reproduce this issue?

**Reply by Pavan-0xblock** [reply]
I don't have too much experience about this type of errors! Can you please elaborate more by what do you mean by MRE?

**Reply by glitch-txs** [reply]
Hi, MRE stands for Minimal Reproducible example. As per the Wagmi's issue template:
> Please provide a link via [new.wagmi.sh](https://new.wagmi.sh/) or a link to a minimal repository that can reproduce the problem you ran into. npm create wagmi@latest can also be used as a starter template.
This makes investigating issues and helping you out significantly easier! For most issues, you will likely get asked to provide one so why not add one now :)
